x-pack/platform/plugins/shared/fleet/common/types/models/cloud_connector.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+export type CloudProvider = 'aws' | 'azure' | 'gcp';
+
+export interface CloudConnectorSecretVarValue {
+  isSecretRef: boolean;
+  id: string;
+}
+
+export interface CloudConnectorRoleArn {
+  type?: 'text';
+  value: string;
+}
+
+export interface CloudConnectorSecretVar {
+  type?: 'password';
```

Comment:
Nit: Referring to this variable as a password seems misleading to me. I understand that only an ID is stored, so I do not have a concern for the data, but this will likely cause confusion to future security-minded reviewers.

Could we use some other notation here, like "secret id" or "password id"? If using "password" here is following a strict paradigm that is already in place, could we at a minimum add a comment explaining it is not storing an actual password?

Reply:
Type here is driven strict schema for input var fields which defined by Fleet package spec for secret fields. I can add comment for security reviewers